Userland/Libraries/LibWeb/Layout/BlockFormattingContext.cpp

```diff
@@ -1077,9 +1077,19 @@ void BlockFormattingContext::layout_floating_box(Box const& box, BlockContainer
     };
 
     // Next, float to the left and/or right
-    if (box.computed_values().float_() == CSS::Float::Left) {
+
+    Optional<FloatSide> float_side;
+
+    // FIXME: Honor writing-mode, direction and text-orientation.
+    if (box.computed_values().float_() == CSS::Float::Left || box.computed_values().float_() == CSS::Float::InlineStart) {
+        float_side = FloatSide::Left;
+    } else if (box.computed_values().float_() == CSS::Float::Right || box.computed_values().float_() == CSS::Float::InlineEnd) {
+        float_side = FloatSide::Right;
+    }
```

Comment:
We resolve other direction-dependent values in StyleComputer, i.e before layout even happens. I think that could work for these as well :)

Reply:
Does that mean defining a new StyleValue subclass?

Reply:
That should not be necessary. We just need `StyleComputer::set_property_expanding_shorthands()` to somehow turn `float: inline-end` into `float: right` or whatever. :)

If you look around that file, you can see how other logical properties are handled, for example in `for_each_property_expanding_shorthands()` (although note that those are logical properties, not logical values)

Reply:
ok I've moved it there